src/sentry/static/sentry/app/views/eventsV2/eventQueryParams.tsx

```diff
@@ -1,58 +1,124 @@
 import {assert} from 'app/types/utils';
 
-export type ColumnValueType =
-  | '*' // Matches to everything
+export type ColumnType =
+  | '*' // Matches to everything TODO(mark) remove this in favour of explicit type lists.
   | 'string'
   | 'integer'
   | 'number'
   | 'duration'
   | 'timestamp'
-  | 'boolean'
-  | 'never'; // Matches to nothing
+  | 'boolean';
+
+export type ColumnValueType = ColumnType | 'never'; // Matches to nothing
+
+export type AggregateParameter = {
+  kind: 'column' | 'value';
+  columnTypes: Readonly<ColumnType[]>;
+  required: boolean;
+};
 
 // Refer to src/sentry/api/event_search.py
 export const AGGREGATIONS = {
   count: {
-    type: '*',
+    parameters: [],
+    outputType: 'number',
+    type: ['*'],
     isSortable: true,
   },
   count_unique: {
-    type: '*',
-    isSortable: true,
-  },
-  /*
-  rpm: {
-    type: 'numeric',
+    parameters: [
+      {
+        kind: 'column',
+        columnTypes: ['string', 'integer', 'number', 'duration', 'timestamp', 'boolean'],
+        required: true,
+      },
+    ],
+    outputType: 'number',
+    type: ['*'],
     isSortable: true,
   },
-  pXX: {
-    type: 'numeric',
-    isSortable: true,
-  },
-  */
   min: {
+    parameters: [
+      {
+        kind: 'column',
+        columnTypes: ['integer', 'number', 'duration', 'timestamp'],
+        required: true,
+      },
+    ],
+    outputType: null,
     type: ['timestamp', 'duration'],
     isSortable: true,
   },
   max: {
+    parameters: [
+      {
+        kind: 'column',
+        columnTypes: ['integer', 'number', 'duration', 'timestamp'],
+        required: true,
+      },
+    ],
+    outputType: null,
     type: ['timestamp', 'duration'],
     isSortable: true,
   },
   avg: {
+    parameters: [
+      {
+        kind: 'column',
+        columnTypes: ['integer', 'number', 'duration'],
+        required: true,
+      },
+    ],
+    outputType: null,
     type: ['duration'],
     isSortable: true,
   },
   sum: {
+    parameters: [
+      {
+        kind: 'column',
+        columnTypes: ['integer', 'number', 'duration'],
+        required: true,
+      },
+    ],
+    outputType: null,
     type: ['duration'],
     isSortable: true,
   },
+  last_seen: {
+    parameters: [],
+    outputType: 'timestamp',
+    type: [],
+    isSortable: true,
+  },
+  p75: {
+    parameters: [],
+    outputType: 'duration',
+    type: [],
+    isSortable: true,
+  },
+  p95: {
+    parameters: [],
+    outputType: 'duration',
+    type: [],
+    isSortable: true,
+  },
+  p99: {
+    parameters: [],
+    outputType: 'duration',
+    type: [],
+    isSortable: true,
+  },
 } as const;
```

Comment:
This is missing latest_event(), apdex(), impact(), error_rate(), rpm() and rps().

Reply:
Yes, I left those off for now and will be adding them in after with the multi-parameter function support.